vmdb/app/models/miq_expression.rb

```diff
@@ -1209,7 +1209,8 @@ def self.quote(val, typ, mode=:ruby)
     case typ.to_s
     when "string", "text", "boolean", nil
       val = "" if val.nil? # treat nil value as empty string
-      return mode == :sql ? ActiveRecord::Base.connection.quote(val) : "'" + val.to_s.gsub(/'/, "\\\\'") + "'" # escape any embedded single quotes
+      # escape any embedded single quotes, etc. - needs to be able to handle even values with trailing backslash
+      return mode == :sql ? ActiveRecord::Base.connection.quote(val) : val.to_s.inspect
```

Comment:
inspect wraps the string in double quotes, but the previous code wrapped in in single quotes.  Won't this break things?